core/controllers/admin.py

```diff
@@ -71,6 +76,114 @@ class AdminHandler(base.BaseHandler):
+                        'update_feature_flag_rules'
+                    ]
+                },
+                'default_value': None
+            },
+            'exploration_id': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'collection_id': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'num_dummy_exps_to_generate': {
+                'schema': {
+                    'type': 'int'
+                },
+                'default_value': None
+            },
+            'num_dummy_exps_to_publish': {
+                'schema': {
+                    'type': 'int'
+                },
+                'default_value': None
+            },
+            'new_config_property_values': {
+                'schema': {
+                    'type': 'object_dict',
+                    'validation_method': (
+                        validation_method.validate_new_config_property_values)
+                },
+                'default_value': None
+            },
+            'config_property_id': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'data': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'topic_id': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'feature_name': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'commit_message': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'new_rules': {
+                'schema': {
+                    'type': 'list',
+                    'items': {
+                        'type': 'dict',
+                        'properties': [{
+                            'name': 'value_when_matched',
+                            'schema': {
+                                'type': 'bool'
+                            }
+                        }, {
+                            'name': 'filters',
+                            'schema': {
+                                'type': 'list',
+                                'items': {
+                                    'type': 'object_dict',
+                                    'object_class': parameter_domain.PlatformParameterFilter # pylint: disable=line-too-long
+                                }
+                            }
+                        }]
```

Comment:
the items in the list are `PlatformParameterRule`? If yes can we use validate method of the domain class?

Reply:
https://github.com/oppia/oppia/blob/f9b874546b62be6c4a24f2b235ccd19a24c3eb0c/core/domain/platform_parameter_domain.py#L520-L524

@BenHenning (as CODEOWNER for the feature_gating project), looks like the validation method of `PlatformParameterRule` class is incomplete i.e, it doesn't validates the `value_when_matched` property, is that expected? [I don't see any docstring for the `value_when_matched` property so I have no context of the type of the property.]

Is it okay to add the validation for the `_value_when_matched` in the validate method? If yes, is this class represents any datastore model(s)? If yes, do we need to audit existing models with the new validation method?

Reply:
Nice catch @DubeySandeep. I don't think any auditing is needed here because we haven't created any platform parameters yet. The type of ``_value_when_matched`` depends on the ``_data_type`` of the parent ``PlatformParameter`` object, so I think that would need to get passed down through ``validate()`` in order to validate that the value is the correct type.

Reply:
Done